{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#metaBEAT analysis of Illumina MiSeq output for Lake project_Cytb_Dec_2016"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Mark sure your metaBEAT image version is the latest\n",
    "\n",
    "`metaBEAT - metaBarcoding and Environmental DNA Analyses tool version: v.0.97.9-global`\n",
    "Update from your terminal\n",
    "\n",
    "####check the image, the latest ID should be f2646ec2d352\n",
    "\n",
    "sudo docker images\n",
    "####if not, use the command to update the image from your terminal\n",
    "\n",
    "sudo docker pull chrishah/metabeat\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###The first step will be to trim/clean our raw Illumina data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working\r\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mkdir trimming"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/trimming\n"
     ]
    }
   ],
   "source": [
    "cd trimming"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__Prepare Querymap__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Prepare a text file specifying the samples to be processed including the format and location of the reads. \n",
    "\n",
    "The below command expects the Illumina data to be present in 2 fastq files (forward and reverse reads) per sample in a directory `../raw_reads/`. It expects the files to be named 'sampleID', followed by '_R1' or '_R2' to identify the forward/reverse read file respectively. \n",
    "\n",
    "The raw data need to be downloaded with `How_to_download_Rawdata_from_SRA.ipynb` (see [here](https://github.com/HullUni-bioinformatics/Li_et_al_2018_eDNA_fish_monitoring/blob/master/Cytb/How_to_download_Cytb_Rawdata_from_SRA.ipynb))\n",
    "\n",
    "SampleID must corresponds to the first column in the file `Sample_accessions.tsv` (see [here](https://github.com/HullUni-bioinformatics/Li_et_al_2018_eDNA_fish_monitoring/blob/master/Cytb/supplementary_data/Sample_accessions.tsv)), marker is `Cytb`.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "\n",
    "%%bash\n",
    "\n",
    "for a in $(cat ../supplementary_data/Sample_accessions.tsv | grep \"Cytb\" | cut -f 1 | grep \"SampleID\" -v)\n",
    "do\n",
    "    R1=$(ls -1 ../raw_reads/$a-Cytb_* | grep \"_R1.fastq\")\n",
    "    R2=$(ls -1 ../raw_reads/$a-Cytb_* | grep \"_R2.fastq\")\n",
    "\n",
    "    echo -e \"$a\\tfastq\\t$R1\\t$R2\"\n",
    "done > Querymap.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The resulting file should look e.g. like below:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BET01\tfastq\t../raw_reads/BET01-Cytb_R1.fastq.gz\t../raw_reads/BET01-Cytb_R2.fastq.gz\r\n",
      "BET02\tfastq\t../raw_reads/BET02-Cytb_R1.fastq.gz\t../raw_reads/BET02-Cytb_R2.fastq.gz\r\n",
      "BET03\tfastq\t../raw_reads/BET03-Cytb_R1.fastq.gz\t../raw_reads/BET03-Cytb_R2.fastq.gz\r\n",
      "BET04\tfastq\t../raw_reads/BET04-Cytb_R1.fastq.gz\t../raw_reads/BET04-Cytb_R2.fastq.gz\r\n",
      "BET05\tfastq\t../raw_reads/BET05-Cytb_R1.fastq.gz\t../raw_reads/BET05-Cytb_R2.fastq.gz\r\n",
      "BET06\tfastq\t../raw_reads/BET06-Cytb_R1.fastq.gz\t../raw_reads/BET06-Cytb_R2.fastq.gz\r\n",
      "BET07\tfastq\t../raw_reads/BET07-Cytb_R1.fastq.gz\t../raw_reads/BET07-Cytb_R2.fastq.gz\r\n",
      "BET08\tfastq\t../raw_reads/BET08-Cytb_R1.fastq.gz\t../raw_reads/BET08-Cytb_R2.fastq.gz\r\n",
      "BET09\tfastq\t../raw_reads/BET09-Cytb_R1.fastq.gz\t../raw_reads/BET09-Cytb_R2.fastq.gz\r\n",
      "BET10\tfastq\t../raw_reads/BET10-Cytb_R1.fastq.gz\t../raw_reads/BET10-Cytb_R2.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "!head Querymap.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Raw data trimming, removal of adapter sequences and merging of readsusing the `metaBEAT` pipeline."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": false
   },
   "source": [
    "__The manual of metaBEAT_global.py can run the command as below__\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!metaBEAT_global.py -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starttime: Tue Dec 12 12:23:46 UTC 2017\n",
      "\n",
      "Endtime: Wed Dec 13 05:20:53 UTC 2017\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "echo -e \"Starttime: $(date)\\n\"\n",
    "\n",
    "metaBEAT_global.py \\\n",
    "-Q Querymap.txt \\\n",
    "--trim_qual 30 \\\n",
    "--trim_minlength 100 \\\n",
    "--merge --product_length 400 --forward_only \\\n",
    "-@ haikuilee@gmail.com \\\n",
    "-n 5 -v &> log_trimming\n",
    "\n",
    "echo -e \"Endtime: $(date)\\n\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__After trimming, there is no queries under sample \"LGSB1\", then it will not been included in \"chimera detection\"__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__Query survival after trimming__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Some stats on the read counts before/after trimming, merging etc. are summarized for you in `metaBEAT_read_stats.csv`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes.AxesSubplot at 0x7f4f74ead710>"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXMAAAEACAYAAABBDJb9AAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAEuBJREFUeJzt3X+MZXdZx/H3tgNidUuzVncroJuIIJBolYhVIT78MsUY\nMJqQoDFUiPEPww8TZRf/sQlBav8QYkj8AwNUNCiRSECMaamcoCgoSYulWCu4awrCFqVlqzHK2vGP\nc6Yznd47c+/Mne/znHPfr+TunO+5597zmTMzz577nHvOBUmSJEmSJEmSJEmSJEmSpJU5tuBy54GL\nwP8BXweeA5wA/hj4zuH+lwMPrjyhJGllztEX751uBt4wTJ8BbmqaSJK0tHPAt+yadw9wcpg+NYwl\nSYX9C3AH8CngF4d5D+y4/9iusSSpoGuGr98K3Ak8j8cW7682TSRJesTGgst9afj6FeBP6Q+AXqBv\nr3yZvtjfP+NxXwS+/ZAZJWndfB546qqf9Arg+DD9TcDHgR+nPwB6Zph/ltkHQDdXHWbEbswOUMiN\n2QEKuTE7QCE3ZgcoZOnaucie+Un6vfGt5f8QuJW+f/4+4NVsvzVR853ODlDI6ewAhZzODlDI6ewA\nY7ZIMT8HXDtj/leBF602jiSpItss2yI7QCGRHaCQyA5QSGQHKKRc7SwXSJJGYOnaedlRpNBMkR2g\nkMgOUEhkBygksgOMmcVckrQv2yyStDzbLJK0jizm7UR2gEIiO0AhkR2gkMgOMGYWc0nSvuyZS9Ly\n7JlL0sFsXKQvortuGxdTYxXhnvm2yA5QSGQHKCSyAxQSyevfhM0Zt5Q65p65JK2jRT/Q+aA2G6xD\nklZhc/YO8bFH/mmbZbl1umcuSRNgMW8nsgMUEtkBConsAIVEdoAxs5hL0gTYM5eknj1zSVIui3k7\nkR2gkMgOUEhkBygksgOMmcVckibAnrkk9eyZS5JyWczbiewAhUR2gEIiO0AhkR1gzCzmkjQB9swl\nqWfPXJKUy2LeTmQHKCSyAxQS2QEKiewAY2Yxl7Rm5n2i0LjZM5e0bvbojdszlyQlspi3E9kBCons\nAIVEdoBCIjvAmFnMJWkC7JlLWjf2zCVJNVnM24nsAIVEdoBCIjtAIZEdYMws5pK0Ri4H7gA+NIxP\nALcB9wK3AlfNedzo34gvaXI2YXPGba/5GRmXs+ie+euAz+5YwVn6Yv404PZhLEkq7MnAR4Dns71n\nfg9wcpg+NYxncc98W2QHKCSyAxQS2QEKiUbrWds987cCvwY8vGPeSeDCMH2B7cIuSUqwsc/9Pwnc\nT98vjznL7Pc/17uB88P0g8CdQDeMt55zHcZdsTyO64zZ5/51GW/NO+r17TG7mxPrSPNsrfSGYXx+\n94pX4TeB+4BzwJeA/wLeQ99WOTUscw22WSSNxyTbLMv4MbZ75jcDZ4bps8BNGYFGJrIDFBLZAQqJ\n7ACFRKP1TLKYL/s+860V3AS8mP6tiS9gfjGXJDXgtVkkrRuvzSJJqsli3k5kBygksgMUEtkBCons\nAGNmMZekCbBnLmnd2DOXJNVkMW8nsgMUEtkBConsAIVEdoAxs5hL0gTYM5e0buyZS5Jqspi3E9kB\nConsAIVEdoBCIjvAmFnMJWkC7JlLWjf2zCVJNVnM24nsAIVEdoBCIjtAIZEdYMws5pI0AfbMJa0b\ne+aSpJos5u1EdoBCIjtAIZEdoJDIDjBmFnNJmgB75pLWjT1zSVJNFvN2IjtAIZEdoJDIDlBIZAcY\nM4u5JE2APXNJ68aeuSSpJot5O5EdoJDIDlBIZAcoJLIDjJnFXJImwJ65pHVjz1ySVJPFvJ3IDlBI\nZAcoJLIDFBLZAcbMYi5JE2DPXNK6sWcuSarJYt5OZAcoJLIDFBLZAQqJ7ABjZjGXpDXwBOCTwJ3A\nZ4G3DPNPALcB9wK3AlfNefysBpQkZdqEzRm3veZnZFy9K4avG8AngOcCNwNvGOafAW5qGUiSDmFt\ni/mWK4C/B54F3AOcHOafGsbNA41MZAcoJLIDFBLZAQqJRuuZZDFfpGd+GX2b5QLwUeBu+kJ+Ybj/\nAtuFXZJU3BPp2yzPBx7Ydd9X5zzGPXNJ1Uxyz3xjiWW/BnwYeDb93vgp4MvANcD9ezzu3cD5YfpB\n+r38bhjH8NWxY8eOW433mN3NeNgjjjJfADcM4/O7V7wKV7P9TpVvBD4GvJD+AOiZYf5ZPAC6iMgO\nUEhkBygksgMUEo3Ws5Z75tcAt9D3zS8D3gPcDtwBvA94Nf3/Ii9fdsWSpNXx2iyS1o3XZpEk1WQx\nbyeyAxQS2QEKiewAhUR2gDGzmEvSBNgzl7Ru7JlLkmqymLcT2QEKiewAhUR2gEIiO8CYWcwlaQLs\nmUtaN/bMJUk1WczbiewAhUR2gEIiO0Ahsbqn2rhIv3c76zZJy1w1UZJG4tLx+XV7mp1fe+aSpmhO\nXxz27o3bM5ckJbKYtxPZAQqJ7ACFRHaAQiI7wJhZzCVpAuyZS5oie+aSpPGxmLcT2QEKiewAhUR2\ngEIiO8CYWcwlaQLsmUuaInvmkqTxsZi3E9kBConsAIVEdoBCIjvAmDW4NssTv/DYeRfPwuYfHP26\nJUmrsAl377q9ZhMuf3N2MEmTtgmbc27z7ttrfkb+5TTYM3/mrvG3Hf0qJWnN2DNvJ7IDFBLZAQqJ\n7ACFRHaAMbOYS9IEWMzb6bIDFNJlByikyw5QSJcdYMws5pI0ARbzdiI7QCGRHaCQyA5QSGQHGDOL\nuSRNgMW8nS47QCFddoBCuuwAhXTZAcbMYi5JE2AxbyeyAxQS2QEKiewAhUR2gDGzmEvSBFjM2+my\nAxTSZQcopMsOUEiXHWDMFinmTwE+CtwNfAZ47TD/BHAbcC9wK3DVUQSUJO1vkWL+deBXgGcB1wG/\nDDwDOEtfzJ8G3D6MNV9kBygksgMUEtkBConsAGO2SDH/MnDnMP2fwD8CTwJeCtwyzL8F+KmVp5Mk\nHYnTwL8Cx4EHdsw/tmu8Zca1gd/k9cwlHbW1u575MgdAvxl4P/A64KEZK874hiVJLP7hFI+jL+Tv\nAT4wzLsAnKJvw1wD3D/7oTfQ79BDf4z03M47Y/jarcF4a7pKnszx1rwqeTLH1wJvK5Qnc/x6+pbu\nip5v3mIw+/6tefOWP/L6cMMwPr97xatyDPh94K275t8MnBmmzwI3zXisbZZtkR2gkMgOUEhkBygk\nVvhca9dmWcRzgYfp/8e8Y7hdT//WxI+w91sTLeaSMqxdMV+kzfLXzO+tv2jZFUqSVs8zQNuJ7ACF\nRHaAQiI7QCGx/EM2LrL9Boydt7Wz6AFQSSro0vHZtftY8yTZ3DNvp8sOUEiXHaCQLjtAIV12gDGz\nmEvSBFjM24nsAIVEdoBCIjtAIZEdYMws5pI0ARbzdrrsAIV02QEK6bIDFNJlBxgzi7kkTYDFvJ3I\nDlBIZAcoJLIDFBLZAcbMYi5JE2Axb6fLDlBIlx2gkC47QCFddoAxs5hL0gRYzNuJ7ACFRHaAQiI7\nQCGRHWDMLOaSNAEW83a67ACFdNkBCumyAxTSZQcYM4u5JE2AxbydyA5QSGQHKCSyAxQS2QHGzGIu\nSRNgMW+nyw5QSJcdoJAuO0AhXXaAMbOYS9IEWMzbiewAhUR2gEIiO0AhkR1gzCzmkjQBFvN2uuwA\nhXTZAQrpsgMU0mUHmG0D+k+N3nXbuJiZareN7ACSVNsl+vq927HjrZPsxT3zdiI7QCGRHaCQyA5Q\nSGQHGDOLuSRNgMW8nS47QCFddoBCuuwAhXTZAcbMYi6pkI2LjOBgY0UW83YiO0AhkR2gkMgOUEjA\npeMzazmXSh1srMhiLkkTYDFvp8sOUEiXHaCQLjtAIV12gDGzmEs6Iva/W7KYtxPZAQqJ7ACFRHaA\no7N0/zsaBZski7kkTYDFvJ0uO0AhXXaAQrrsAIV02QHGzGIuSROwSDF/J3ABuGvHvBPAbcC9wK3A\nVauPNjmRHaCQyA5QSGQHKCSyA4zZIsX8XcD1u+adpS/mTwNuH8aSpOJO8+g983uAk8P0qWE8yyZs\n7rq9aRMuf/MRZpVUw4y//82tt7Ss6jF7LD9r/l73HWT+kVn6uQ/aMz9J33ph+Hpyj2UlSUdsFR9O\nsc//UDfQ79hD31o/t/POGL52azDemq6SJ3O8Na9KnszxtcDb6uS57MPw8BXMdBnw8Iz5Gw/BpStn\nPB+zf8yP2H3H6x99986YG8ClffZWdy6/13je8lvzllmeWGLFe42DvlgCnOcIneaxbZZTw/Q12GZZ\nRGQHKCSyAxQS2QF2OWB7Ypnnmrt87P2YVcy3zbLbB4FXDtOvBD5wwOdZJ112gEK67ACFdNkBCumy\nA4zZIsX8vcDfAE8H7gN+AbgJeDH9WxNfMIwlSUkW6Zm/Ys78F60yyBoI3PPYErgttgRuiy2RHWDM\nPANUWkvzrmiosVrFu1m0mC47QCFddoBCupzVbl3RcLdjzZPs0GWufOzcM5ekCbCYtxPZAQqJ7ACF\nRHaAQiI7wJhZzCVpAizm7XTZAQrpsgMU0mUHKKTLDjBmFnNJmgCLeTuRHaCQyA5QSGQHKCSyA4yZ\nxVyStK85F9ra+B9mnrCwcTE1rbQ+VnyBqmXXsapcXmhrS9JJQ5ceP+eEhePNo0jSBNhmaSeyAxQS\n2QEKiewAhUR2gDGzmEvSBFjM2+myAxTSZQc4uHkXqDrw8Z5uRcGmoMsOMGZeaEtaytwLVHm8R6nc\nM28nsgMUEtkBConsAIVEdoAxs5hL0gTYZmmnyw5QSJcdoJBuNU+zcbFvAT1m/kNw6crVrGNVNoBL\nfhDGilnMpUkYUy//EvPPiUn9cIxRs83STmQHKCSyAxQS2QHq6LIDjJrFXJImwGLeTpcdoJAuO0Ah\nXXaAOiI7wKhZzCVpAizm7UR2gEIiO0AhkR2gji47wKhZzCVpAizm7XTZAQrpsgMU0mUHqCOyA4xa\nsWK+ATMvYuQHV+gw5l0cq+rv1cov5nXE5v7dqqFixXzrZIJZt1lnt41KZAcoJNqubuuEmpK/V/HY\nWfPypmedY97f7bK6FWZaP8WKuSTpICzm7XTZAQrpsgMU0mUHqCOyA4yaxVySJmBExXzeQZaNeQe2\nDnDAa9kDT0stH/t8fyM76HUosT2558HJMR20hIN9H7GadSz9mIK67ACjNqKrJs670tqxZefvcRBp\n2SvPrfJKdWO66t0qzfu+4WA/wyz7XQlwFd/HQa42uNffjaZkRHvmo9dlByikyw5QSJcdoI7IDjBq\nFnNJmoDDFvPrgXuAfwbOHD5OC2l9xxiea16PeIWOtPe/4nUcxNzjJyvspTf5OXH06xiTLjvAqB2m\nmF8OvJ2+oD8TeAXwjFWEOlp7nZi07GOWcu3wXHNOCFmlZU86OchJKodZx1s53Pc97+exypNqmvyc\nePS2WPNazp3ZAUbtMMX8OcDngPPA14E/Al62gkxTdVV2gDoezA5QiNtim9viMA5TzJ8E3Ldj/IVh\nniSpscO8NXHB14Mv+Nqjx+eeAHzDIdY7VqezA9RxPjtAIeezAxRyPjvAqB3mzabXATfS98wB3gg8\nDPzWjmU+B3zXIdYhSevo88BTW61sY1jhaeDx9EcvRnAAVJK020uAf6LfA39jchZJkiRpvS1y4tDv\nDPd/Gvj+Rrmy7Lc9fo5+O/wD8HHge9tFa2rRE8p+kP4N1z/dIlSSRbZFAHcAn2HaZ8/sty2uBv6C\nvm37GeCGZsnaeidwAbhrj2Wa1s3L6Vssp4HHMbtv/hPAnw/TPwR84qhDJVpke/ww8MRh+nqmuT0W\n2Q5by/0l8GfAz7QK19gi2+Iq4G7gycP46lbhGltkW9wIvGWYvhr4D0Z1QcCFPY++QM8r5kvXzcOe\nzr/IiUMvBW4Zpj9J/4t78pDrrWqR7fG3wNbbNT/J9h/wlCx6QtlrgD8BvtIsWXuLbIufBd5Pf64G\nwL+3CtfYItviS8CVw/SV9MX8UqN8Lf0V8MAe9y9dNw9bzBc5cWjWMlMsYLD8iVSvZvt/3ylZ9Pfi\nZcDvDuOpnse+yLb4buAE8FHgU8DPt4nW3CLb4h3As4B/o28vvK5NtHKWrpuHffmy6B/g7vezT/UP\nd5nv6/nAq4AfPaIsmRbZDm8Dzg7LHmO6F9heZFs8DvgB4IXAFfSv3j5B3y+dkkW2xa/Tt1+C/hyV\n24DvAx46ulhlLVU3D1vMvwg8Zcf4KWy/VJy3zJOHeVO0yPaA/qDnO+h75nu91BqrRbbDs+lfZkPf\nG30J/UvvDx55urYW2Rb30bdW/nu4fYy+gE2tmC+yLX4EePMw/XngHPB0+lcs66R53VzkxKGdjfzr\nmOYBvy2LbI/voO8bXtc0WVvLnlD2Lqb7bpZFtsX3AB+hP0B4Bf1BsWe2i9jMItvit4HfGKZP0hf7\nE43ytXaaxQ6ANqubs04c+qXhtuXtw/2fpn85OWX7bY/foz+oc8dw+7vWARtZ5Pdiy5SLOSy2LX6V\n/h0tdwGvbZqurf22xdXAh+hrxV30B4en6L30xwX+l/6V2atY77opSZIkSZIkSZIkSZIkSZIkSZIk\nabf/Bx9mbuNjeQbMAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7f4f74ead3d0>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%matplotlib inline\n",
    "import pandas as pd\n",
    "\n",
    "df = pd.read_csv('metaBEAT_read_stats.csv',index_col=0)\n",
    "df['fraction'] = df['queries']/(df['total']*0.5)\n",
    "df.fraction.hist(bins=50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working\n"
     ]
    }
   ],
   "source": [
    "cd ../"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###Next stage of the processing is chimera detection and removal of putative chimeric sequences\n",
    "\n",
    "We'll do that using `uchime` as implemented in `vsearch`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mkdir chimera_detection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/chimera_detection\n"
     ]
    }
   ],
   "source": [
    "cd chimera_detection"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__Prepare REFmap__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Convert reference database from Genbank to fasta format to be used in chimera detection.\n",
    "\n",
    "Prepare Refmap file, i.e. text file that specifies the location and the format of the reference to be used.\n",
    "\n",
    "The reference sequences in Genbank format should already be present in the `supplementary_data` directory: `Cytb_evohull_DEC_2017.gb`\n",
    " and  `Astatotilapia_calliptera.gb` is positive control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Write REFmap\n",
    "for file in $(ls -1 ../supplementary_data/reference_DBs/* | grep \"gb$\")\n",
    "do\n",
    "      echo -e \"$file\\tgb\"\n",
    "done > REFmap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!cat REFmap.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__Transfer reference DB from gb to fasta format__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This will produce `refs.fasta`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "metaBEAT_global.py \\\n",
    "-R REFmap.txt \\\n",
    "-f \\\n",
    "-@ haikuilee@gmail.com"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!head -2 refs.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "__Now run chimera detection__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "echo -e \"Starttime: $(date)\\n\"\n",
    "\n",
    "for a in $(cut -f 1 ../trimming/Querymap.txt)\n",
    "do\n",
    "    if [ -s ../trimming/$a/$a\\_trimmed.fasta ]\n",
    "    then\n",
    "        echo -e \"\\n### Detecting chimeras in $a ###\\n\"\n",
    "        mkdir $a\n",
    "        cd $a\n",
    "        vsearch --uchime_ref ../../trimming/$a/$a\\_trimmed.fasta --db ../refs.fasta \\\n",
    "        --nonchimeras $a-nonchimeras.fasta --chimeras $a-chimeras.fasta &> log\n",
    "        cd ..\n",
    "\n",
    "    else\n",
    "        echo -e \"$a is empty\"\n",
    "    fi\n",
    "done\n",
    "\n",
    "echo -e \"Endtime: $(date)\\n\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Last step is taxonomic assignment of reads based on a BLAST - LCA approach using the metaBEAT pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working\n"
     ]
    }
   ],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "!mkdir non-chimeras"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/working/non-chimeras\n"
     ]
    }
   ],
   "source": [
    "cd non-chimeras/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__Prepare Querymap and Refmap txt files__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__To skip the `GLOBAL` folder under chimeras detection, add a new line in below command  if [ \"$a\" != \"GLOBAL\" ]__\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "#Querymap\n",
    "for a in $(ls -l ../chimera_detection/ | grep \"^d\" | perl -ne 'chomp; @a=split(\" \"); print \"$a[-1]\\n\"')\n",
    "do\n",
    "   if [ \"$a\" != \"GLOBAL\" ]\n",
    "   then\n",
    "      echo -e \"$a-nc\\tfasta\\t../chimera_detection/$a/$a-nonchimeras.fasta\"\n",
    "   fi\n",
    "done > Querymap.txt\n",
    "\n",
    "\n",
    "\n",
    "#Write REFmap\n",
    "for file in $(ls -1 ../supplementary_data/reference_DBs/* | grep \"gb$\")\n",
    "do\n",
    "      echo -e \"$file\\tgb\"\n",
    "done > REFmap.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!head -10  Querymap.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!cat REFmap.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__Sequence clustering and taxonomic assignment via metaBEAT__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Starttime: Fri Dec 15 14:54:17 UTC 2017\n",
      "\n",
      "Endtime: Fri Dec 15 16:26:16 UTC 2017\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "echo -e \"Starttime: $(date)\\n\"\n",
    "\n",
    "metaBEAT_global.py \\\n",
    "-Q Querymap.txt \\\n",
    "-R REFmap.txt \\\n",
    "--cluster --clust_match 1 --clust_cov 3 \\\n",
    "--blast --min_ident 0.95 \\\n",
    "-m Cytb -n 5 \\\n",
    "-E -v \\\n",
    "-@  haikuilee@gmail.com \\\n",
    "-o Cytb_LP-trim_30-merged-nonchimeras-cluster_1c3-blast-min_ident_0.95 &> log_non-chimeras\n",
    "\n",
    "echo -e \"Endtime: $(date)\\n\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__Results are under the `GLOBAL` folder__\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
